pkg/analysis/markerscope/config.go

```diff
@@ -0,0 +1,595 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package markerscope
+
+import (
+	"maps"
+
+	"sigs.k8s.io/kube-api-linter/pkg/markers"
+)
+
+// ScopeConstraint defines where a marker is allowed to be placed using bit flags.
+type ScopeConstraint uint8
+
+const (
+	// FieldScope indicates the marker can be placed on fields.
+	FieldScope ScopeConstraint = 1 << iota
+	// TypeScope indicates the marker can be placed on type definitions.
+	TypeScope
+
+	// AnyScope indicates the marker can be placed on either fields or types.
+	AnyScope = FieldScope | TypeScope
+)
```

Comment:
We don't use this kind of config for public APIs. Use a string type with PascalCase values